### DISCUSSION TITLE
How to build DragonflyDB from source in Mac?

### DISCUSSION THREAD

**Discussion by Abhra303** [dragonflydb/dragonfly#1166] [Q&A] [answered]
Hi, I am Abhradeep Chakraborty. I am trying to build the dragonflydb from source as I want to work on the #1158 issue. But seems like it requires Linux (src- https://github.com/dragonflydb/dragonfly/blob/main/docs/build-from-source.md). Is there any way to build DragonflyDB for Mac?

I can't use the docker-compose here because I want my changes to appear in dragonflyDB build. I am considering two ways here - 1. install ubuntu virtual machine and then build
2. Run a linux distribution container and then develop in the container

Is their any simpler way to do that?

My Mac info -

```bash
$ uname -a
Darwin Abhradeeps-MacBook-Pro.local 22.4.0 Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:41 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T8103 arm64
```

**Comment by romange** [comment, marked as answer]
Hi @Abhra303 . The project is indeed for linux only. Maybe some day it will work for MacOS :)

I suggest going for option (2) and use `ghcr.io/romange/ubuntu-dev:20` like we do in our CI tests: 
https://github.com/dragonflydb/dragonfly/actions/runs/4843063121/workflow#L38

**Reply by Abhra303** [reply]
Thanks!

**Comment by ThatOneCalculator** [comment]
On this note, eventually supporting FreeBSD would be awesome!
